dynawo/sources/ModelicaCompiler/Scripts_OMC_1_13_2/factory.py

```diff
@@ -2508,6 +2578,73 @@ def prepare_for_literalconstants(self):
 
                 self.list_for_literalconstants.append(var)
 
+    ##
+    # prepare the lines that constitues the body for evalCalculatedVars
+    # @param self : object pointer
+    # @return
+    def prepare_for_evalcalculatedvars(self):
+        index = 0
+        for var, expr in self.list_calculated_vars.items():
+            self.list_for_evalcalculatedvars.append("  calculatedVars[" + str(index)+"] /* " + var.get_name() + "*/ = " + expr+";\n")
+            index += 1
+
+
+    ##
+    # return the list of lines that constitues the body of evalCalculatedVars
+    # @param self : object pointer
+    # @return list of lines
+    def get_list_for_evalcalculatedvars(self):
+        return self.list_for_evalcalculatedvars
+
+    ##
+    # prepare the lines that constitues the body for evalCalculatedVars
+    # @param self : object pointer
+    # @return
+    def prepare_for_evalcalculatedvari(self):
+        index = 0
+        for var, expr in self.list_calculated_vars.items():
+            self.list_for_evalcalculatedvari.append("  if (iCalculatedVar == " + str(index)+")  /* "+ var.get_name() + " */\n")
+            self.list_for_evalcalculatedvari.append("    return "+ expr+";\n")
+            index += 1
+        self.list_for_evalcalculatedvari.append("  return 0;\n")
+
+
+    ##
+    # return the list of lines that constitues the body of evalCalculatedVars
+    # @param self : object pointer
+    # @return list of lines
+    def get_list_for_evalcalculatedvari(self):
+        return self.list_for_evalcalculatedvari
+
+    ##
+    # prepare the lines that constitues the body for evalCalculatedVars
+    # @param self : object pointer
+    # @return
+    def prepare_for_evaljcalculatedvar(self):
+        self.list_for_evaljcalculatedvar.append("  // not needed\n")
```

Comment:
Currently we only have 2 different types of calculated variable in a modelica model:
- constant reals
- real aliased on a discrete variable (real = discrete)
Thus there is no value in the Jacobian for those cases.
This will appear if we detect more advanced calculated variables, with the shape of var = f(syst variables)